src/main/CommandHandler.cpp

```diff
@@ -396,6 +404,86 @@ CommandHandler::generateLoad(std::string const& params, std::string& retStr)
     }
 }
 
+void
+CommandHandler::benchmark(std::string const& params, std::string& retStr)
+{
+    if (!mApp.getConfig().ARTIFICIALLY_GENERATE_LOAD_FOR_TESTING)
+    {
+        retStr = "Set ARTIFICIALLY_GENERATE_LOAD_FOR_TESTING=true in "
+                 "the stellar-core.cfg if you want this behavior";
+        return;
+    }
+
+    uint32 createAccounts = 0;
+    uint32_t nAccounts = 1000000;
+    uint32_t txRate = 1000;
+    uint32_t duration = 60 * 10;
+
+    std::map<std::string, std::string> map;
+    http::server::server::parseParams(params, map);
+
+    if (!parseNumParam(map, "createaccounts", createAccounts, retStr,
+                       Requirement::OPTIONAL_REQ))
+    {
+        retStr = "Invalid value for the parameter 'createaccounts'";
+        return;
+    }
+
+    if (createAccounts > 0)
+    {
+        Benchmark::BenchmarkBuilder builder(mApp.getNetworkID());
+        builder.setNumberOfInitialAccounts(createAccounts)
+            .populateBenchmarkData()
+            .createBenchmark(mApp);
+
+        retStr = fmt::format("{{\"createdAccounts\": {:d}}}", createAccounts);
+        return;
+    }
+
+    if (!parseNumParam(map, "accounts", nAccounts, retStr,
+                       Requirement::OPTIONAL_REQ))
+    {
+        retStr = "Invalid value for the parameter 'accounts'";
+        return;
+    }
+
+    if (!parseNumParam(map, "txrate", txRate, retStr,
+                       Requirement::OPTIONAL_REQ))
+    {
+        retStr = "Invalid value for the parameter 'txrate'";
+        return;
+    }
+
+    if (!parseNumParam(map, "duration", duration, retStr,
+                       Requirement::OPTIONAL_REQ))
+    {
+        retStr = "Invalid value for the parameter 'duration'";
+        return;
+    }
+
+    Benchmark::BenchmarkBuilder builder(mApp.getNetworkID());
+    builder.setNumberOfInitialAccounts(nAccounts)
+        .setTxRate(txRate)
+        .initializeBenchmark();
+
+    mApp.getMetrics().NewMeter({"benchmark", "run", "started"}, "run").Mark();
+
+    auto stopCallback = [this](Benchmark::Metrics metrics) {
+
+        mApp.getMetrics()
+            .NewMeter({"benchmark", "run", "complete"}, "run")
+            .Mark();
+        BenchmarkReporter().reportBenchmark(metrics, mApp.getMetrics(),
```

Comment:
is there any reason for this class to exists?
reportBenchmark could just be a free function and that would work

Reply:
I had two different reporting objects - normal and more verbose. Only one is left, so this class is no longer needed. Thank you for noticing.